packages/portfolio-contract/test/rebalance.test.ts

```diff
@@ -0,0 +1,357 @@
+import test from 'ava';
+import { AmountMath } from '@agoric/ertp';
+import type { Brand, Amount } from '@agoric/ertp/src/types.js';
+import { Far } from '@endo/marshal';
+import { planRebalanceFlow } from '../src/plan-solve.js';
+import type { AssetPlaceRef } from '../src/type-guards-steps.js';
+
+// Shared Tok brand + helper
+const { brand: TOK_BRAND } = (() => ({ brand: Far('Tok') as Brand<'nat'> }))();
+const token = (v: bigint) => AmountMath.make(TOK_BRAND, v);
+const ZERO = token(0n);
+
+/**
+ * Graph / network model specification (declarative)
+ *
+ * Nodes:
+ *   Pools: 'Aave_Arbitrum', 'Beefy_re7_Avalanche', 'Compound_Ethereum'
+ *   Chain hubs: '@Arbitrum', '@Avalanche', '@Ethereum', '@noble', '@agoric'
+ *   Agoric local leaves: '+agoric', '<Deposit>', '<Cash>'
+ *
+ * Edges (conceptual attributes):
+ *   - Intra-chain (leaf <-> hub): variableFee=1, time=1 (seconds)
+ *   - Noble <-> Agoric: variableFee=2, time=10 (seconds)
+ *   - EVM hub -> Noble (CCTP): time=1080 (18 minutes), variableFee≈0
+ *   - Noble -> EVM hub (CCTP return): time=20 (seconds), variableFee≈0
```

Comment:
`CCTP return` could be different directions for different perspectives, consider just use `CCTP`